src/Lucid/Relations/BaseRelation.js

```diff
@@ -115,7 +115,7 @@ class BaseRelation {
    * @private
    */
   _validateRead () {
-    if (!this.$primaryKeyValue || !this.parentInstance.$persisted) {
+    if (typeof this.$primaryKeyValue === 'undefined' || !this.parentInstance.$persisted) {
```

Comment:
I believe we should remove the `undefined` check too.

Reply:
Also it's always better to check agains `void 0` than `undefined`

Reply:
I check for if fkey column exist. If not exist - throw error. Still remove?